### ISSUE TITLE
Configure sidebars for singlehtml

### ISSUE THREAD

**Issue by davidism** [sphinx-doc/sphinx#5246]
It's common to configure one set of sidebars for the index and another set for all other pages. For example, the index page will show project links in the sidebar and the full table of contents in the body, while other pages will show the local table of contents and navigation.

```python
html_sidebars = {
    'index': [
        'project.html',
        'versions.html',
        'searchbox.html',
    ],
    '**': [
        'localtoc.html',
        'relations.html',
        'versions.html',
        'searchbox.html',
    ]
}
```

When using the singlehtml builder, the page name is always "index". It would be desirable to show the table of contents in the sidebar for this builder, since due to #2589 the table of contents is replaced by the pages and won't show up in the body. But since the page name is "index", it uses the index sidebars configured for the regular html builder.

A solution to this would be to add a `singlehtml_sidebars = []` configuration, and fall back to the `html_sidebars` configuration if it's not present. It would also be good to be able to configure the default in `theme.conf` like how `html_sidebars` works.

**Comment by davidism** [comment]
I implemented this using the `builder-inited` event in an extension. https://github.com/pallets/pallets-sphinx-themes/blob/c9f4461d575dfbc4deb3deaebab89e4ba03c4576/pallets_sphinx_themes/__init__.py#L62-L68

If the builder is an instance of the singlehtml builder and the config is set, it replaces `html_config['index']`.

I'd have to dig into more Sphinx internals to figure out how this would be implemented natively.

**Comment by tk0miya** [comment]
Good idea! I just implemented it as #5261.
